### ISSUE TITLE
Implementation of avatars is too data-heavy

### ISSUE THREAD

**Issue by f-r00t** [kryptokrona/hugin-api#185]
Some thoughts:

1. Currently the data in the cache is 87% avatar data, and only 13% is "essential" data. In the long run, this won't scale, especially since each message from the same address stores that user's avatar individually in every message. This could be mitigated by having a specific database table with avatars for each known user, albeit quite data heavy in itself. I propose that we generate avatars on request instead (i.e. when posts?avatars=true).

2. There are currently at least a hundred known users of Hugin Mobile, who already use Hugin Cache to sync boards data. This recent change leads to mobile users downloading almost 8 times more data for no benefit, which might put a dent into some peoples mobile data quota. This is because the avatar change made avatars to be included as the default option. I would highly recommend making it the other way around, i.e. avatars should only be included upon request (whether they are generated on the fly, or taken from the database). Of course, we could just update Hugin Mobile, but seeing as some of our users are not connected to Google Play, autoupdates may not work for everyone, causing some people to be stuck with the data drain and having no way of mitigating it other than uninstalling the app, and for that reason I reckon that we should change the api anyway.

TL;DR:
1. Don't store avatars in DB. Just generate them when somebody specifically asks for avatars=true.
2. Make avatars=false the default option.

**Comment by mjovanc** [comment]
Makes sense!

**Comment by f-r00t** [comment]
Nice! Also forgot:

3. Change excludeAvatars=false to avatars=true, because double negatives are hard to grasp during late night coding sessions 😝

**Comment by mjovanc** [comment]
@TechyGuy17 Currently we have made a hot-fix to just simply disable avatar from JSON response as per default. But you can set with query params "excludeAvatar=false" to include them. But we will remove them from the model. So the comment above is still required.

**Comment by TechyGuy17** [comment]
Alright will get on it ASAP again

**Comment by mjovanc** [comment]
Great, just when you have time. Currently it's no stress. It works for the developers so. 👍🏻

**Comment by mjovanc** [comment]
@TechyGuy17 I will take over this issue since I will have time for it.
